prov/rxm/src/rxm_domain.c

```diff
@@ -313,22 +313,25 @@ static int rxm_query_collective(struct fid_domain *domain,
 				uint64_t flags)
 {
 	struct rxm_domain *rxm_domain;
+	int ret;
 
 	rxm_domain = container_of(domain, struct rxm_domain,
 				  util_domain.domain_fid);
 
 	if (!rxm_domain->util_coll_domain)
 		return -FI_ENOSYS;
 
+	if (rxm_domain->offload_coll_domain)
+		ret = fi_query_collective(rxm_domain->offload_coll_domain,
+					  coll, attr, flags);
+	else
+		ret = -FI_ENOSYS;
+
+	if (FI_SUCCESS == ret || flags & OFI_OFFLOAD_PROV_ONLY)
```

Comment:
Please use forward logic.

Who will set the OFI_OFFLOAD_PROV_ONLY flag?

Reply:
Done.